backend/compact-connect/tests/smoke/military_affiliation_smoke_tests.py

```diff
@@ -0,0 +1,131 @@
+# ruff: noqa: S101  we use asserts for smoke testing
+import os
+import sys
+import time
+from datetime import UTC, datetime
+
+import requests
+
+# This script is used to test the military affiliations upload flow of the Compact Connect API.
+# It is currently configured to run locally against a sandbox environment. To run this script, you must
+# set the following constant values to match your environment
+# This should include the https:// prefix  (ex. https://sandbox.compactconnect.org)
+API_URL = os.environ['API_URL']
+# The cognito ID token issued when the user is authenticated.
+TEST_USER_COGNITO_ID_TOKEN = os.environ['TEST_USER_ID_TOKEN']
+
+
+def test_military_affiliation_upload():
+    # Step 1: Create a military affiliation record in the DB using the POST
+    # '/v1/provider-users/me/military-affiliation' endpoint.
+    # Step 2: Get the S3 pre-signed URL from the response body and use it to upload a test pdf file to the S3 bucket.
+    # Step 3: Verify that the test pdf file was uploaded successfully by checking the response status code.
+    # Step 4: Get the provider data from the GET '/v1/provider-users/me' endpoint and verify that the military
+    # affiliation record is active.
+    headers = {
+        'Authorization': 'Bearer ' + TEST_USER_COGNITO_ID_TOKEN,
+    }
+
+    post_body = {
+        'fileNames': ['military_affiliation.pdf'],
+        'affiliationType': 'militaryMember',
+    }
+    post_api_response = requests.post(
+        url=API_URL + '/v1/provider-users/me/military-affiliation', headers=headers, json=post_body, timeout=10
+    )
+
+    assert (
+        post_api_response.status_code == 200
+    ), f'Failed to POST military affiliations record. Response: {post_api_response.json()}'
+
+    """
+    Response body should include S3 pre-sign url form in this format:
+    {
+    ...
+        'documentUploadFields': [
+            {
+                'fields': {
+                    'key': f'compact/{TEST_COMPACT}/provider/{provider_id}/document-type/military-affiliations'
+                    f'/{today}/1234#military_affiliation.pdf',
+                    'x-amz-algorithm': 'AWS4-HMAC-SHA256',
+                },
+                'url': 'https://provider-user-bucket.s3.amazonaws.com/',
+            }
+        ],
+        'fileNames': ['military_affiliation.pdf'],
+        'status': 'initializing',
+    }
+    """
```

Comment:
nit: I'm used to this format of docstring only being used for function, class, or module comments. Would it be more appropriate to use `#` like you do at the start of this function?

Reply:
In this case, I was intending to have a multi-line comment, not a docstring. Looking it up I see the convention for multi line strings is single quotes. I'll update this to reflect that.

Reply:
Good to know!